### PR TITLE
Switch to pyproject.toml build system and migrate versioneer to new build system.

### LINKED ISSUE #149: requirements.txt is maybe not being used, and extras are definitely not being used
I believe we need to deprecate the requirements.txt files and specify dependencies in pyproject.toml like so: 

https://stackoverflow.com/questions/6237946/optional-dependencies-in-distutils-pip

I'll take care of this.

### PR DESCRIPTION
**Pull request by pbeaucage**
Fixes #149, `pip install -e .[bluesky,ui]` now installs the correct requirements.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,74 @@
...
+    "astropy",
+    "fabio",
+    "nodejs",
+    "silx==2.0.0",
+    "pygix",
+    "pydata_sphinx_theme",
+    "numexpr<2.8.5",
+]
+
+[project.optional-dependencies]
```
Comment (martintb) [inline pyproject.toml:44]: Three more optional dependency sets to consider
[doc] - sphinx, documentation building, etc
[test] - pytest, black, codecov, pylint, etc
[all] - union of everything

**Review by martintb — COMMENTED** [review]
LGTM

Consider the extra optional build sets that I mention

File: pyproject.toml (hunk 1 above)
Comment (pbeaucage) [reply pyproject.toml:44]: Great suggestion, done, thx!

**Review by martintb — APPROVED** [review]
LGTM

**Pull request merged**